### PR TITLE
Fixes #75

### LINKED ISSUE #75: Need an LP and NP Type Definition
In the constructor of MAB, we use a Union of LPs and Union of NPs written explicitly. 

https://github.com/fidelity/mabwiser/blob/master/mabwiser/mab.py#L732

This is problematic.. 

This creates an issue for ALSN not being able to a generic LP Type or NP Type, unless it also explicity states all options as a Union -- which is not good. 

This issue is to create a Type Defintion, LPType and NPType similar to ARM type  https://github.com/fidelity/mabwiser/blob/master/mabwiser/utils.py#L13

This can simplify the MAB() constuctor using these type. And... ALNS and other libraries on top, can refer to these types without necessarily knowing what is in it. When mabwiser introduces new policies, pip upgrading the library at the top-level library will uptake new types automatically. 

Note: once this features is complete, it also requires a minor code change in the ALNS to refer to these new types in the MABSelector()

### PR DESCRIPTION
**Pull request by AshishPvjs**
Code for fixing the Issue #75

### CODE REVIEW

File: mabwiser/mab.py [hunk 1]
```diff
@@ -667,6 +667,25 @@ def _is_compatible(self, learning_policy: LearningPolicy):
                                                 LearningPolicy.ThompsonSampling))
 
 
+LearningPolicyType = NewType('LearningPolicyType', Union[LearningPolicy.EpsilonGreedy,
```
Comment (bkleyn) [inline mabwiser/mab.py:670]: Mind adding a brief comment like "LearningPolicyType is the union of all learning policies"

File: mabwiser/mab.py [hunk 2]
```diff
@@ -667,6 +667,25 @@ def _is_compatible(self, learning_policy: LearningPolicy):
...
+                                                         LearningPolicy.Random,
+                                                         LearningPolicy.Softmax,
+                                                         LearningPolicy.ThompsonSampling,
+                                                         LearningPolicy.UCB1,
+                                                         LearningPolicy.LinGreedy,
+                                                         LearningPolicy.LinTS,
+                                                         LearningPolicy.LinUCB])
+
+
+NeighborhoodPolicyType = NewType('NeighborhoodPolicyType', Union[None,
```
Comment (bkleyn) [inline mabwiser/mab.py:681]: Also add comment for NeighborhoodPolicyType

**Comment by bkleyn** [comment]
Note code in `tests/test_base.py` lines 78 and 82 that might be possible to replace with the new types.

**Review by skadio — APPROVED** [review]
Looks great, and I assume all tests pass?

**Review by skadio — APPROVED** [review]

**Review by skadio — DISMISSED** [review]

**Review by skadio — APPROVED** [review]

**Pull request merged**